src/Interactive/EditorFeatures/Core/Implementation/Interactive/Commands/ResetCommand.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Interactive;
+using Microsoft.VisualStudio.Language.StandardClassification;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.InteractiveWindow;
+using Microsoft.VisualStudio.InteractiveWindow.Commands;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Interactive
+{
+    [Export(typeof(IInteractiveWindowCommand))]
+    [InteractiveWindowRole(InteractiveWindowRoles.Any)]
+    internal sealed class ResetCommand : InteractiveWindowCommand
+    {
+        private const string CommandName = "reset";
+        private const string NoConfigParameterName = "noconfig";
+        private readonly IStandardClassificationService _registry;
+
+        [ImportingConstructor]
+        public ResetCommand(IStandardClassificationService registry)
+        {
+            _registry = registry;
+        }
+
+        public override string Description
+        {
+            get { return "Reset the execution environment to the initial state, keep REPL history."; }
+        }
+
+        public override IEnumerable<string> Names
+        {
+            get { yield return CommandName; }
+        }
+
+        public override string CommandLine
+        {
+            get { return "[" + NoConfigParameterName + "]"; }
+        }
+
+        public override IEnumerable<KeyValuePair<string, string>> ParametersDescription
+        {
+            get
+            {
+                return new ReadOnlyCollection<KeyValuePair<string, string>>(new[]
```

Comment:
It looks like it's just to avoid the caller from casting and mutating the array.  We could yield instead.